Add our own compare dialog with editable text to replace monaco-editor

TomNUSDS opened the pull request:
<img width="1385" alt="image" src="https://user-images.githubusercontent.com/74203452/158674751-87be7020-ce9c-4404-818e-ef21b29b5f9a.png">

- Fixes #4734
- Replace `monaco-editor` with our own.
- Hidden behind feature flag until tested in staging. Set a flag `showNewDiffEditor` to test!
- Lots of refactor cleanup around dialogs and passing data around.
- Fixes around `showNotification` when combined with `<Alert>` (pain to track down)

 Includes:
 - Text differ compare (with tests)
 - Side-by-side scrolling of content
 - Live edit the content to make corrections.
 - Should work on staging/production without needing to lower security.

## Pull reviewers stats
Stats for the last 30 days:
|                                                                                                                                                                            | User               | Total reviews | Median time to review                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | Total comments |
| -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------- |
| <a href="https://github.com/MauriceReeves-usds"><img src="https://avatars.githubusercontent.com/u/74194189?u=f81b3f069b28469e1fb39322ba3c17112c1dd1d0&v=4" width="20"></a> | MauriceReeves-usds | **49**        | [58m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74194189~n~'MauriceReeves-usds)~p~30~r~(~(d~'r7eow1~t~'179)~(d~'r7f0jk~t~'ge)~(d~'r7gfjf~t~'1i7f)~(d~'r7gkdc~t~'1km)~(d~'r7gkkx~t~'1n8x)~(d~'r7gklk~t~'1su)~(d~'r7exg7~t~'1ca)~(d~'r7j9ud~t~'47)~(d~'r7j6x9~t~'j)~(d~'r7j93e~t~'2a)~(d~'r7pvw5~t~'8me)~(d~'r7rk5m~t~'g1)~(d~'r7ii8y~t~'7c4)~(d~'r7ixn7~t~'ao6)~(d~'r7ix8d~t~'2ip)~(d~'r7gkoh~t~'2pr)~(d~'r7k3ul~t~'i)~(d~'r7kh87~t~'269)~(d~'r7kfak~t~'2nx)~(d~'r7vu7z~t~'6ox)~(d~'r7tgos~t~'1mh)~(d~'r7vuee~t~'czah)~(d~'r7gugi~t~'5fh)~(d~'r7ij1v~t~'8m3)~(d~'r7ixv9~t~'4129)~(d~'r7j502~t~'dl)~(d~'r7iz0o~t~'97)~(d~'r7j81e~t~'4v)~(d~'r7j5s6~t~'eh)~(d~'r7j43v~t~'ez)~(d~'r7j4kk~t~'v5)~(d~'r7j7q8~t~'12)~(d~'r7j7bl~t~'db)~(d~'r7rke9~t~'1cyr)~(d~'r7rk82~t~'14lw)~(d~'r7t8iu~t~'10xu)~(d~'r7trsx~t~'3mi)~(d~'r86rtt~t~'2ou)~(d~'r7psct~t~'7i7o)~(d~'r810pm~t~'5mv4)~(d~'r851ga~t~'5u)~(d~'r82l0j~t~'aj9)~(d~'r82wm3~t~'23z5)~(d~'r7vuas~t~'23)~(d~'r7rkat~t~'17zq)~(d~'r7ggj1~t~'1aw4)~(d~'r814x6~t~'1fy)~(d~'r7vu74~t~'51q)~(d~'r82xbj~t~'24ob)~(d~'r8foaf~t~'3g)~(d~'r8hfp7~t~'2of)~(d~'r8hfw8~t~'6nm)~(d~'r8fogk~t~'29)~(d~'r8l2yp~t~'2rh)~(d~'r8nh45~t~'5ht)))) | 15             |
| <a href="https://github.com/carlosfelix2"><img src="https://avatars.githubusercontent.com/u/63804190?u=009d26bf9914817d2f9a066811541d0f6f3155c9&v=4" width="20"></a>       | carlosfelix2       | 21            | [15h 31m](https://app.flowwer.dev/charts/review-time/~(u~(i~'63804190~n~'carlosfelix2)~p~30~r~(~(d~'r7ejkt~t~'nz)~(d~'r7gnjg~t~'1zzm)~(d~'r7pn9i~t~'1qax)~(d~'r7ik32~t~'o3)~(d~'r7io0t~t~'c1)~(d~'r7po11~t~'7dvw)~(d~'r7tpv5~t~'1pgz)~(d~'r86krr~t~'9z5)~(d~'r82kk3~t~'a2t)~(d~'r813my~t~'5q)~(d~'r810wn~t~'58xu)~(d~'r86arq~t~'4nw)~(d~'r8hfrz~t~'8p)~(d~'r8hg7g~t~'1rw9)~(d~'r8ftzo~t~'4tw)~(d~'r86ad6~t~'1fde)~(d~'r8jabx~t~'1r)~(d~'r8qxgg~t~'837)~(d~'r8sg3m~t~'1d20)~(d~'r8spql~t~'44)~(d~'r8sg1r~t~'1de7)~(d~'r8sps0~t~'1n4g)~(d~'r8sptb~t~'fd)~(d~'r8l4ux~t~'1l0w)~(d~'r8lezb~t~'3jt)~(d~'r82koo~t~'18p8)~(d~'r86bj6~t~'3q13)~(d~'r8hfho~t~'18ax)~(d~'r8shfd~t~'173b)~(d~'r8shg4~t~'1742)~(d~'r8shko~t~'178m))))                                                                                                                                                                                                                                                                                                                                                                                                                                                           | 37             |
| <a href="https://github.com/jimduff-usds"><img src="https://avatars.githubusercontent.com/u/62258514?u=b7ce81d1478c472fb5bf6f3d31edb1a4454d55dd&v=4" width="20"></a>       | jimduff-usds       | 18            | [4h 29m](https://app.flowwer.dev/charts/review-time/~(u~(i~'62258514~n~'jimduff-usds)~p~30~r~(~(d~'r7f5ep~t~'172)~(d~'r7gi6p~t~'mq)~(d~'r7f4rn~t~'7fn)~(d~'r7i9je~t~'1jyb)~(d~'r88hjp~t~'156)~(d~'r7ttqp~t~'1tcj)~(d~'r82kv2~t~'ads)~(d~'r80rpq~t~'4xh1)~(d~'r7iabk~t~'34on)~(d~'r82juw~t~'bz)~(d~'r82kxp~t~'1k0u)~(d~'r8fpbf~t~'1de)~(d~'r86yh5~t~'k39)~(d~'r8hqad~t~'109)~(d~'r8se4l~t~'1bh1)~(d~'r8htmt~t~'ekz)~(d~'r8t1iu~t~'adv)~(d~'r8lpyp~t~'ej7)~(d~'r8r04j~t~'1j)~(d~'r8qk66~t~'4vzj))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | 11             |
| <a href="https://github.com/cwinters-usds"><img src="https://avatars.githubusercontent.com/u/86251310?u=e7f46bd48ca250dea43f9e0d6433a7fb8627ed30&v=4" width="20"></a>      | cwinters-usds      | 18            | [2h 35m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86251310~n~'cwinters-usds)~p~30~r~(~(d~'r7ep8o~t~'76)~(d~'r7d2e4~t~'co)~(d~'r7eoyf~t~'1el)~(d~'r7gxzs~t~'l)~(d~'r7ep3q~t~'16e4)~(d~'r7ep6f~t~'16gt)~(d~'r7ep71~t~'16hf)~(d~'r7euc3~t~'1bmh)~(d~'r7euh4~t~'1bri)~(d~'r7rx2c~t~'f1o)~(d~'r7tf4o~t~'3sv)~(d~'r7rwtb~t~'aic)~(d~'r86fkz~t~'8u)~(d~'r86kuo~t~'a22)~(d~'r7z318~t~'321r)~(d~'r80zp4~t~'74v)~(d~'r80zrk~t~'77f)~(d~'r82qoa~t~'1prf)~(d~'r82w52~t~'353)~(d~'r8e0s4~t~'93tu)~(d~'r8e0wt~t~'9mu)~(d~'r8e13g~t~'6f)~(d~'r8e26t~t~'fs)~(d~'r8e14q~t~'9um)~(d~'r8e0s4~t~'93tu)~(d~'r8ncmk~t~'108)~(d~'r8ncol~t~'129)~(d~'r8jtis~t~'47o)~(d~'r8jtk6~t~'492)~(d~'r8ju7e~t~'4wa))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | 9              |
| <a href="https://github.com/whytheplatypus"><img src="https://avatars.githubusercontent.com/u/410846?v=4" width="20"></a>                                                  | whytheplatypus     | 15            | [16h 44m](https://app.flowwer.dev/charts/review-time/~(u~(i~'410846~n~'whytheplatypus)~p~30~r~(~(d~'r7f1qd~t~'35p)~(d~'r7f1hq~t~'45q)~(d~'r7f2yw~t~'5mw)~(d~'r7q2q3~t~'35b)~(d~'r7tfcz~t~'416)~(d~'r86doc~t~'1bq)~(d~'r86b47~t~'1fvo)~(d~'r86cbb~t~'1h2s)~(d~'r7v69p~t~'1h1x)~(d~'r7v9xd~t~'182)~(d~'r7v9y7~t~'18w)~(d~'r7v6wj~t~'1bv4)~(d~'r7vafl~t~'1fe6)~(d~'r7vcl4~t~'16y)~(d~'r7gaky~t~'14y1)~(d~'r7t83r~t~'16f5)~(d~'r7v5wh~t~'1crv)~(d~'r8hfq4~t~'h2)~(d~'r8sfxy~t~'1bjt)~(d~'r8shps~t~'1dbn)~(d~'r8si1i~t~'1dnd)~(d~'r8srxc~t~'sd)~(d~'r8qo6p~t~'6vux)~(d~'r8l00d~t~'1g6c)~(d~'r8qnz2~t~'4zsf)~(d~'r8kzrv~t~'1agr)~(d~'r8l5ex~t~'1g3t)~(d~'r8l6cx~t~'1h1t)~(d~'r8hg25~t~'q2))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | 30             |
| <a href="https://github.com/TomNUSDS"><img src="https://avatars.githubusercontent.com/u/74203452?u=6eccca154e9d5c808c5269d9604648aec3c9a412&v=4" width="20"></a>           | TomNUSDS           | 15            | [1h 6m](https://app.flowwer.dev/charts/review-time/~(u~(i~'74203452~n~'TomNUSDS)~p~30~r~(~(d~'r7pzxh~t~'260)~(d~'r7pzz6~t~'27p)~(d~'r7glx3~t~'40a)~(d~'r7tquw~t~'bd7)~(d~'r7iino~t~'7l2)~(d~'r7exfw~t~'gn)~(d~'r7glca~t~'1lu0)~(d~'r7glm1~t~'1m3r)~(d~'r7glns~t~'1m5i)~(d~'r7ici4~t~'3czu)~(d~'r7icli~t~'3d38)~(d~'r7icth~t~'3db7)~(d~'r82zy3~t~'1v10)~(d~'r84q27~t~'xf)~(d~'r84r0n~t~'1vv)~(d~'r84t4m~t~'3zu)~(d~'r84thl~t~'4ct)~(d~'r88cg2~t~'6x)~(d~'r88dkz~t~'z7)~(d~'r7w1h5~t~'ho)~(d~'r7ttc5~t~'2z4)~(d~'r7txr4~t~'2pp)~(d~'r7ty7m~t~'367)~(d~'r7pzwc~t~'3i0)~(d~'r8g3pi~t~'2w)~(d~'r8g3ps~t~'36)~(d~'r8g58l~t~'6v)~(d~'r85nw7~t~'fez)~(d~'r86ufu~t~'dq)~(d~'r8sn1l~t~'1wf))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | **43**         |
| <a href="https://github.com/ahay-agile6"><img src="https://avatars.githubusercontent.com/u/19178435?u=aa7d4ddd31665f83e677e20967502557741345e4&v=4" width="20"></a>        | ahay-agile6        | 14            | [2h 44m](https://app.flowwer.dev/charts/review-time/~(u~(i~'19178435~n~'ahay-agile6)~p~30~r~(~(d~'r7eubf~t~'11c)~(d~'r7d0hl~t~'cx)~(d~'r7q2zb~t~'57u)~(d~'r7tnwd~t~'8eo)~(d~'r7s620~t~'dg2)~(d~'r7tgzs~t~'2tp)~(d~'r7pshp~t~'78sx)~(d~'r7idms~t~'3e4i)~(d~'r7tx7b~t~'6sm)~(d~'r88fpu~t~'ix)~(d~'r86kjk~t~'efq)~(d~'r7xcej~t~'v57)~(d~'r8e46d~t~'8pm)~(d~'r8skx3~t~'32x))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         | 2              |
| <a href="https://github.com/oslynn"><img src="https://avatars.githubusercontent.com/u/20068283?u=bcad2a3d00ffe648463e3fbe1894762aeab72404&v=4" width="20"></a>             | oslynn             | 12            | [56m](https://app.flowwer.dev/charts/review-time/~(u~(i~'20068283~n~'oslynn)~p~30~r~(~(d~'r7ejcv~t~'g1)~(d~'r7f0w4~t~'sy)~(d~'r7fbck~t~'b9)~(d~'r7grbj~t~'2ai)~(d~'r7iez3~t~'1t0m)~(d~'r7ir48~t~'2zh)~(d~'r7ir6a~t~'31j)~(d~'r7r4gx~t~'our)~(d~'r81f1p~t~'1s)~(d~'r8311x~t~'2vq)~(d~'r84rmx~t~'hu)~(d~'r8hfwj~t~'kg)~(d~'r8r9tm~t~'4ht)~(d~'r8uhru~t~'2g30))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     | 3              |
| <a href="https://github.com/jorg3lopez"><img src="https://avatars.githubusercontent.com/u/49923512?v=4" width="20"></a>                                                    | jorg3lopez         | 11            | [1h 19m](https://app.flowwer.dev/charts/review-time/~(u~(i~'49923512~n~'jorg3lopez)~p~30~r~(~(d~'r7ema3~t~'3d9)~(d~'r7vw9t~t~'1y8)~(d~'r7tced~t~'3vd)~(d~'r88mbb~t~'2tj)~(d~'r88q3d~t~'5u)~(d~'r86mkn~t~'4wy)~(d~'r854kz~t~'3aj)~(d~'r854xt~t~'3nd)~(d~'r854yg~t~'3o0)~(d~'r85514~t~'3qo)~(d~'r8556w~t~'3wg)~(d~'r8560y~t~'a8)~(d~'r86rje~t~'amf)~(d~'r7rpog~t~'1ddd)~(d~'r7rpqg~t~'1dfd)~(d~'r88q3d~t~'5u)~(d~'r839rp~t~'3l1)~(d~'r8hmnt~t~'7bq))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | 7              |
| <a href="https://github.com/jbiskie"><img src="https://avatars.githubusercontent.com/u/84460447?v=4" width="20"></a>                                                       | jbiskie            | 9             | [21m](https://app.flowwer.dev/charts/review-time/~(u~(i~'84460447~n~'jbiskie)~p~30~r~(~(d~'r7ep3v~t~'2d)~(d~'r7gsis~t~'1v2e)~(d~'r7vueg~t~'10)~(d~'r7rza1~t~'6b)~(d~'r82x6m~t~'19j)~(d~'r8houx~t~'za)~(d~'r8li4o~t~'re)~(d~'r8r3t6~t~'13x)~(d~'r8udja~t~'1bc8))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | 0              |
| <a href="https://github.com/brick-green-agile6"><img src="https://avatars.githubusercontent.com/u/86254221?u=894ef5a4773dd01e92be595af8f1879041f85002&v=4" width="20"></a> | brick-green-agile6 | 9             | [16h 35m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86254221~n~'brick-green-agile6)~p~30~r~(~(d~'r7ejgq~t~'1520)~(d~'r7vrzr~t~'1ztt)~(d~'r7tc2g~t~'3jg)~(d~'r80vrx~t~'4z7n)~(d~'r7tck5~t~'1a1p)~(d~'r810r3~t~'591)~(d~'r7vsb9~t~'2b7i)~(d~'r8dtai~t~'4y0v)~(d~'r8r437~t~'1fp)~(d~'r8su9g~t~'690)~(d~'r8ugq9~t~'2f1f))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | 4              |
| <a href="https://github.com/kevinhaube"><img src="https://avatars.githubusercontent.com/u/4022304?u=c0f80e2ad03386621394d1830dc33b7b5cc23680&v=4" width="20"></a>          | kevinhaube         | 8             | [1h 6m](https://app.flowwer.dev/charts/review-time/~(u~(i~'4022304~n~'kevinhaube)~p~30~r~(~(d~'r7ei86~t~'1bhy)~(d~'r84v2o~t~'be)~(d~'r86qq2~t~'2c5)~(d~'r86rge~t~'32h)~(d~'r86s5f~t~'1a)~(d~'r86s6m~t~'2h)~(d~'r86snb~t~'h3)~(d~'r7w1vd~t~'cpz)~(d~'r8fro6~t~'28o)~(d~'r86bjt~t~'132l)~(d~'r86cxb~t~'14g3)~(d~'r86hro~t~'19ag)~(d~'r8frpw~t~'290)~(d~'r8qnbt~t~'4c1d)~(d~'r8k9ky~t~'pqx))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | 6              |
| <a href="https://github.com/TurkeyFried"><img src="https://avatars.githubusercontent.com/u/2624105?u=675395bdf4c1b121eb2f291338079089107daf74&v=4" width="20"></a>         | TurkeyFried        | 7             | [1h 32m](https://app.flowwer.dev/charts/review-time/~(u~(i~'2624105~n~'TurkeyFried)~p~30~r~(~(d~'r7iw4u~t~'a8a)~(d~'r831xn~t~'1okt)~(d~'r7s0qn~t~'1uu0)~(d~'r8hfmn~t~'dl)~(d~'r8hh3b~t~'4gw)~(d~'r84v2s~t~'30)~(d~'r8jtb9~t~'405)~(d~'r8ju5r~t~'rm))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | 2              |
| <a href="https://github.com/rhood23699"><img src="https://avatars.githubusercontent.com/u/86322826?v=4" width="20"></a>                                                    | rhood23699         | 7             | [1h 5m](https://app.flowwer.dev/charts/review-time/~(u~(i~'86322826~n~'rhood23699)~p~30~r~(~(d~'r7ejbv~t~'f1)~(d~'r7jvt1~t~'rm9)~(d~'r7vrd6~t~'303)~(d~'r7gra1~t~'290)~(d~'r8l5s3~t~'9q)~(d~'r8qom5~t~'4wnr)~(d~'r8r5bg~t~'9cu2))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | 0              |
| <a href="https://github.com/sethdarragile6"><img src="https://avatars.githubusercontent.com/u/92405130?u=b36435069928f7e5f03109c9f478cdc7158fc01e&v=4" width="20"></a>     | sethdarragile6     | 6             | [3h 56m](https://app.flowwer.dev/charts/review-time/~(u~(i~'92405130~n~'sethdarragile6)~p~30~r~(~(d~'r7gy13~t~'1w)~(d~'r7goc7~t~'6fe)~(d~'r86mg1~t~'5rgg)~(d~'r86oyd~t~'b86)~(d~'r80wgo~t~'4f7c)~(d~'r8jug4~t~'am3))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             | 0              |
| <a href="https://github.com/bgantick"><img src="https://avatars.githubusercontent.com/u/640182?u=901de0e5fa2ff06da055ee0949aa806a6c82dcb8&v=4" width="20"></a>             | bgantick           | 5             | [1h 36m](https://app.flowwer.dev/charts/review-time/~(u~(i~'640182~n~'bgantick)~p~30~r~(~(d~'r7v9mv~t~'2ak)~(d~'r7tstc~t~'dbn)~(d~'r7icnt~t~'3d5j)~(d~'r7tx10~t~'6mb)~(d~'r7txem~t~'63)~(d~'r8fr6x~t~'1f2))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | 4              |
| <a href="https://github.com/meckila"><img src="https://avatars.githubusercontent.com/u/68879427?u=f820e48d2924ecd35b62030423c5a6377fc4e1f4&v=4" width="20"></a>            | meckila            | 5             | [4h 12m](https://app.flowwer.dev/charts/review-time/~(u~(i~'68879427~n~'meckila)~p~30~r~(~(d~'r7q3cx~t~'5lg)~(d~'r7s9hf~t~'gvh)~(d~'r7iq7s~t~'8zt)~(d~'r7txba~t~'2r)~(d~'r7txg7~t~'7o)~(d~'r8rdzy~t~'bn7)~(d~'r8s0ua~t~'yhj)~(d~'r8t8l1~t~'nfv)~(d~'r8t8l8~t~'ng2))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              | 14             |
| <a href="https://github.com/jeremy-page"><img src="https://avatars.githubusercontent.com/u/54326889?u=600f8a60859e9b0ca4634227f00b2cd9bee5e9b8&v=4" width="20"></a>        | jeremy-page        | 3             | [18h 23m](https://app.flowwer.dev/charts/review-time/~(u~(i~'54326889~n~'jeremy-page)~p~30~r~(~(d~'r7cl2e~t~'6i)~(d~'r7iosa~t~'2wp5)~(d~'r8shpj~t~'1f1z))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | 1              |
| <a href="https://github.com/JosiahSiegel"><img src="https://avatars.githubusercontent.com/u/5522990?v=4" width="20"></a>                                                   | JosiahSiegel       | 2             | [16m](https://app.flowwer.dev/charts/review-time/~(u~(i~'5522990~n~'JosiahSiegel)~p~30~r~(~(d~'r7clni~t~'rm)~(d~'r84kbp~t~'q1))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | 0              |
| <a href="https://github.com/snesm"><img src="https://avatars.githubusercontent.com/u/94193373?v=4" width="20"></a>                                                         | snesm              | 2             | [23h 48m](https://app.flowwer.dev/charts/review-time/~(u~(i~'94193373~n~'snesm)~p~30~r~(~(d~'r888gz~t~'1x56)~(d~'r88e8q~t~'1r43))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                | 0              |
| <a href="https://github.com/acoushawk"><img src="https://avatars.githubusercontent.com/u/9059393?u=a28896651bbe093372aa593ebc66296fa638250c&v=4" width="20"></a>           | acoushawk          | 1             | [**7m**](https://app.flowwer.dev/charts/review-time/~(u~(i~'9059393~n~'acoushawk)~p~30~r~(~(d~'r8e5rg~t~'c4)~(d~'r8e5rg~t~'c4))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  | 0              |
| <a href="https://github.com/RickHawesUSDS"><img src="https://avatars.githubusercontent.com/u/48692522?u=323d76e6f867d8ed0b3e56d7e5808313bfcc2ccf&v=4" width="20"></a>      | RickHawesUSDS      | 1             | [51m](https://app.flowwer.dev/charts/review-time/~(u~(i~'48692522~n~'RickHawesUSDS)~p~30~r~(~(d~'r7gvby~t~'2d1))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | 0              |
| <a href="https://github.com/clediggins-usds"><img src="https://avatars.githubusercontent.com/u/89804087?u=5b03415b2bc7766f96908dd40f522633187e4b02&v=4" width="20"></a>    | clediggins-usds    | 1             | [17h 4m](https://app.flowwer.dev/charts/review-time/~(u~(i~'89804087~n~'clediggins-usds)~p~30~r~(~(d~'r7eu3w~t~'1bea))))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | 1              |

TomNUSDS commented on frontend-react/src/components/Spinner.tsx:
```diff
@@ -1,30 +1,19 @@
-        <div
-            style={props.fullPage ? fullPageStyles : defaultStyles}
-            className="grid-container"
-        >
```
Yeah, directly setting `style` is bad.

sethdarragile6 commented on frontend-react/src/utils/diff.ts:
```diff
@@ -0,0 +1,183 @@
+        }
+    };
+
+    const snake = (k: number, p: number, pp: number): number => {
```
🐍

TomNUSDS replied:
Heh... yeah, from the paper:

<img width="595" alt="image" src="https://user-images.githubusercontent.com/74203452/158904825-6c62d101-4cfb-4cf2-a222-4d31e9d0c1dc.png">

sethdarragile6 commented:
Lookin' pretty good, @TomNUSDS ! Found a couple update bugs, tho

1. When editing the top-level org info, the diff shows correctly but when I hit Save, the dialog doesn't go away but I can see that "Saving" toast coming up behind it. Seems to save the change, but NOT any change that I manually make in the right hand diff window. Also, errors seem to be false-flagged as success. For example if I make County Name blank and my Jurisdiction is COUNTY, an error should report (but it correctly does _not_ update.
2. It seems hit-or-miss that changes I make manually on the right hand diff window will actually update on Sender/Receiver updates. It will correctly show what I changed on the previous screen AND update that change correctly, but ignore the manual change I might make

TomNUSDS commented:
> @sethdarragile6 
> 1. When editing the top-level org info, the diff shows correctly but when I hit Save, the dialog doesn't go away but I can see that "Saving" toast coming up behind it. 

This might be a z-index layout issue. The save can be pretty slow so I was trying to leave the dialog up until it was done.

> Seems to save the change, but NOT any change that I manually make in the right hand diff window. Also, errors seem to be false-flagged as success. For example if I make County Name blank and my Jurisdiction is COUNTY, an error should report (but it correctly does _not_ update).

I'll test out error handling on save!

> 2. It seems hit-or-miss that changes I make manually on the right hand diff window will actually update on Sender/Receiver updates. It will correctly show what I changed on the previous screen AND update that change correctly, but ignore the manual change I might make.

This is fairly complicated. Can you check these steps and provide values to help me reproduce it?

So steps:
1. Edit any Sender Settings and change the field: ___a___ to value "___b___"
2. Click Preview save button.
3. In the right-hand-side, edit the json to change ___c___ to value "___d___"
4. Click Save.
5. Refresh the Sender Setting and observer ___e___
Result: Changed made in step 1 were saved but not in step 3.

TomNUSDS commented:
OK. I think I fixed all your issues. GREAT bugs BTW!

sethdarragile6 reviewed (approved):
Looks good! Thanks for taking this on and finally getting this feature to the finish line! 🎉
